### DISCUSSION TITLE
Bills - Sum (Euro) (active and expected bills only) not correct

### DISCUSSION THREAD

**Discussion by nick2k3** [firefly-iii/firefly-iii#8543] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The field "Sum (Euro) (active and expected bills only)" of one group reports a wrong value, not taking in consideration the bill already paid.

<img width="1189" alt="Screenshot 2024-02-14 alle 08 07 08" src="https://github.com/firefly-iii/firefly-iii/assets/42848864/e4479094-8772-43c5-b555-856b9f851233">

### Debug information

Debug information generated at 2024-02-14 08:12:56 for Firefly III version **v6.1.9**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.9 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>957, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + Europe/Brussels</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>info, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has reconciled">:ledger:</span> <span title="Has rules">:wrench:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-02-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-02-29 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.3 Safari/605.1.15</td></tr></tbody></table>

### Expected behaviour

The total sum of the above bills in this month (they are all monthly) is:
156+
92,99+
150 =
__
398,99

The field "Sum (Euro) (active and expected bills only)" should report the sum of the two bills not yet paid:

156+92,99 = 248,99

### Steps to reproduce

Not sure how to reproduce, the bills in other groups have the correct amount.

**Comment by JC5** [comment]
This is not a bug, but expected behavior. The "next expected match" for your paid bill is February 17th, so the currently paid amount (before the 17th) falls into the previous period. Firefly III still expects the bill to be paid on or after the 17th. Since that date lies between 1-29 Feb, it's part of the expected amount.

**Comment by nick2k3** [comment]
hmm, I see.. so I should set the expected date of each bill a bit sooner than the actual billing date, so that they will surely fall into this month?

**Comment by JC5** [comment]
Exactly :)
